prime-router/docs/design/proposals/0023-condition-to-code-mapping/0023-condition-to-code-mapping.md

```diff
@@ -0,0 +1,383 @@
+
+For the below resource:
+
+```json
+  {
+    "resourceType": "Observation",
+    "id": "d683b42a-bf50-45e8-9fce-6c0531994f09",
+    "status": "final",
+    "code": {
+      "coding": [
+       {
+          "system": "http://loinc.org",
+          "code": "80382-5"
+        }
+      ],
+      "text": "Flu A"
+    }
+  }
+```
+the below element in the default transformation utilizes LookupTableValueSet to use key "80382-5" to search Observation-Mapping table stored in the database to return a value from the condition_code column
+
+```yaml
+- name: test-condition
+  resource: 'Bundle.entry.resource.ofType(Observation)'
+  condition: '%resource.code.coding.exists()'
+  bundleProperty: '%resource.meta.tag.code'
+  value: ['%resource.code.coding.code']
+  valueSet:
+    lookupTable:
+      tableName: Observation-Mapping
+      keyColumn: code
+      valueColumn: condition_code
+```
+
+In the example table below we can see that there is no match found for code "80382-5"
+
+Example Observation Mapping Table:
+
+| Member OID                     | Name                                                 | Code    | Descriptor                                                                          | Code System | Version | Status | Condition Name                                                 | Condition Code   | Condition Code System | Condition Code System Version | Value Source |
+|--------------------------------|------------------------------------------------------|---------|-------------------------------------------------------------------------------------|-------------|---------|--------|----------------------------------------------------------------|------------------|-----------------------|-------------------------------|--------------|
+| 2.16.840.1.113762.1.4.1146.798 | Influenza (Tests for influenza A virus Nucleic Acid) | 80588-7 | Influenza virus A M gene [Presence] in Nasopharynx by NAA with probe detection      | LOINC       | 2.74    | Active | Infection caused by novel Influenza A virus variant (disorder) | 541000000000000  | SNOMEDCT              | 2023-03                       | RCTC         |
+| 2.16.840.1.113762.1.4.1146.799 | Influenza (Tests for influenza A virus Antigen)      | 88904-8 | Influenza virus A Ag [Presence] in Lower respiratory specimen by Immunofluorescence | LOINC       | 2.74    | Active | Infection caused by novel Influenza A virus variant (disorder) | 541000000000000  | SNOMEDCT              | 2023-03                       | RCTC         |
+|                                | Influenza - (ABC TESTING LABS)                       | 123456  | Influenza virus A                                                                   | LOCAL       |         | Active | Infection caused by novel Influenza A virus variant (disorder) | 541000000000000  | SNOMEDCT              | 2023-03                       | LOCAL        |   
+
+Example Action Log Error Message: "Missing mapping for: 123456 for table: Observation-Mapping"
```

Comment:
Check if this applies to other places in the doc, but we need to update this example to include "code system" as well since "code" is not enough.

Reply:
Need to think about this one since that is a complete change to how the lookuptables works. Honestly, probably not worth the effort and would make more sense to create a custom function that runs with the condition filter in that case. Brings up the question of what we do when two or more local codes have the same value and don't map/can't to a LOINC code (extreme edge case). Honestly in that case we should probably tell them to ask Regenstrief to create a LOINC code for them.